File: 15295/f21/4/B.py
intput = lambda : map(int, input().split())
n, m, k = intput()
A = list( intput() )
L = n-m+1
pre = [0]
for x in A:
    pre.append(x + pre[-1])
S = [pre[i+m]-pre[i] for i in range(L)]
dp = [[0] * (k+1) for _ in range(L+1)]

for j in range(1,k+1):
    for i in range(1,L+1):
        dp[i][j] = max(dp[i-1][j], S[i-1])
        if i < m: continue
        dp[i][j] = max(dp[i][j], S[i-1] + dp[i-m][j-1])

# for d in dp:
#     print(d)

print( dp[-1][-1] )

File: 15295/f21/4/B.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

int main() {
    int n, m, k;
    cin >> n >> m >> k;
    int L = n - m + 1;
    vector<ll> A(n), pre(n+1), S(L);
    vector<vector<ll>> dp(L+1, vector<ll>(k+1));
    for (int i = 0; i < n; ++i)
        cin >> A[i];
    for (int i = 1; i <= n; ++i)
        pre[i] = pre[i-1] + A[i-1];
    for (int i = 0; i < L; ++i)
        S[i] = pre[i+m] - pre[i];;

    for (int j = 1; j <= k; ++j)
        for (int i = 1; i <= L; ++i) {
            dp[i][j] = max(dp[i-1][j], S[i-1]);
            if (i < m) continue;
            dp[i][j] = max(dp[i][j], S[i-1] + dp[i-m][j-1]);
        }

    cout << dp[L][k] << endl;

}
